Ambiguous overload for var-arg overloaded method.

```
./src/dotty/runtime/Arrays.scala:18: error: Ambiguous overload. The overloaded alternatives of method newInstance in object Array$ with types
 (x$0: Class[_], x$1: Int)Object
 (x$0: Class[_], x$1: Int*)Object
both match arguments (Class[_](clazz), => Int(xs.length))
    val arr = java.lang.reflect.Array.newInstance(clazz, xs.length).asInstanceOf[Array[T]]
```

--- a/src/dotty/tools/dotc/typer/Applications.scala
+++ b/src/dotty/tools/dotc/typer/Applications.scala
@@ -849,7 +849,10 @@ trait Applications extends Compatibility { self: Typer =>
           case tp @ ExprType(tp1) => tp.derivedExprType(repeatedToSingle(tp1))
           case _ => if (tp.isRepeatedParam) tp.argTypesHi.head else tp
         }
-        isApplicable(alt2, tp1.paramTypes map repeatedToSingle, WildcardType) ||
+        val formals1 =
+          if (tp1.isVarArgsMethod && tp2.isVarArgsMethod) tp1.paramTypes map repeatedToSingle
+          else tp1.paramTypes
+        isApplicable(alt2, formals1, WildcardType) ||
         tp1.paramTypes.isEmpty && tp2.isInstanceOf[MethodOrPoly]
       case _ =>
         tp2 match {
@@ -878,7 +881,7 @@ trait Applications extends Compatibility { self: Typer =>
     def winsOwner2 = isDerived(owner2, owner1)
     def winsType2  = isAsSpecific(alt2, tp2, alt1, tp1)
 
-    implicits.println(i"isAsGood($alt1, $alt2)? $tp1 $tp2 $winsOwner1 $winsType1 $winsOwner2 $winsType2")
+    overload.println(i"isAsGood($alt1, $alt2)? $tp1 $tp2 $winsOwner1 $winsType1 $winsOwner2 $winsType2")
 
     // Assume the following probabilities:
     //
@@ -1013,6 +1016,7 @@ trait Applications extends Compatibility { self: Typer =>
     if (isDetermined(candidates)) candidates
     else narrowMostSpecific(candidates) match {
       case result @ (alt1 :: alt2 :: _) =>
+//        overload.println(i"ambiguous $alt1 $alt2")
         val deepPt = pt.deepenProto
         if (deepPt ne pt) resolveOverloaded(alts, deepPt, targs)
         else result

--- a/tests/pos/varargs.scala
+++ b/tests/pos/varargs.scala
@@ -10,4 +10,8 @@ object varargs {
   g(Nil: _*)
   g(1)
   g()
+
+  def f(x: Int): Unit = ()
+  def f(x: Int*): Unit = ()
+  f(1)
 }

